src/install/lockfile.zig

```diff
@@ -733,7 +741,17 @@ pub const Tree = struct {
                 this.queue.deinit();
                 this.sort_buf.deinit(this.allocator);
 
-                return dependency_ids;
+                // take over the `builder.list` pointer for only trees
+                if (@intFromPtr(trees.ptr) != @intFromPtr(list_ptr)) {
+                    var new: [*]Tree = @ptrCast(list_ptr);
+                    bun.copy(Tree, new[0..trees.len], trees);
```

Comment:
We can use this memory because the other field `.dependencies` was already appended to another list and won't be used again